src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -68,52 +98,190 @@ export const updateTrackStatesAndSave: ActionCreator<
   browserStorageService.saveTrackStates(trackStates);
 };
 
+export const getTracksStateInfo = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+) => {
+  const tracksInfo: TrackConfigs = {};
+
+  tracksInfo['gene-focus'] = {
+    showSeveralTranscripts: false,
+    showTranscriptIds: false,
+    showTrackName: false,
+    showFeatureLabel: true,
+    trackType: TrackType.GENE
+  };
+
+  trackCategories.forEach((category) => {
+    category.track_list.forEach((track) => {
+      const trackId = track.track_id.replace('track:', '');
+      const trackType = getTrackType(trackId);
+      if (trackType === TrackType.GENE) {
+        tracksInfo[trackId] = {
+          showSeveralTranscripts: false,
+          showTranscriptIds: false,
+          showTrackName: false,
+          showFeatureLabel: true,
+          trackType
+        };
```

Comment:
here and on line 128 — you probably should create functions that return default track configs:

```
const getDefaultGeneTrackConfig = (): GeneTrackConfig => ({ ... });
const getDefaultRegularTrackConfig = (): RegularTrackConfig => ({ ... })

```

Reply:
✅